cpu/samd21/cmsis/samr21/include/cpu_conf.h

```diff
@@ -20,7 +20,7 @@
 
 #include "cpu_conf_common.h"
 
-#include "samd21.h"
+#include "samr21.h"
```

Comment:
For (i) it is not an option to avoid extra folder depth. The files in cpu/samd21/cmsis are straight and unmodified copied from the Atmel (ASF) distribution. It would be unwise to restructure that.

For (ii), keeping cpu/samd21/include/cpu_conf.h is possible by doing this

    #if defined(__SAMR21G18A__) || defined(__ATSAMR21G18A__)
      #include "../cmsis/samr21/include/samr21g18a.h"
    #elif defined(__SAMD21J18A__) || defined(__ATSAMD21J18A__)
      #include "../cmsis/samd21/include/samd21j18a.h"
    #else
      #error "Unsupported SAMx21 variant."
    #endif

instead of the

    #include "samd21.h"

And only add those variant include when we need them. In other words keep the list small so that the file remains readable.

Alternatively (ii) we can introduce an extra include, samx21.h with all this and more. That file could be generated with a shell script from CMSIS samd21.h and samr21.h.